dependency-injection/DependencyInjection/Controller.cs

```diff
@@ -3,12 +3,28 @@
 namespace DependencyInjection;
 
 [ApiController]
-public class Controller(Singleton _singleton)
+public class Controller(IServiceProvider _serviceProvider, Singleton _singleton)
```

Comment:
let's move `_singleton` to the action it is being used, and showcase `[FromServices]` usage there